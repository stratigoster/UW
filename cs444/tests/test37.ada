PACKAGE foo IS
  type data;
  type pdata is access data;
  type data is
    record
      i:integer;
      s:string;
      p:pdata;
    end record;
  function boo (a,b,c:integer;d,e,f:integer;g,h,i:boolean;k:data) return pdata;
  function plus (a,b:integer) return integer;
  function hi (a:integer) return plus;
--                               ^^^^
-- functions can only return types
-- ie no higher order functions in Ada :(
END foo;
