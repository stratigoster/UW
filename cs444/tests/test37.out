***ERROR***:12:38:	Undefined return type

------
Compilation error
------
